tests/xdc/nexus_request_forwarding_test.go

```diff
@@ -0,0 +1,199 @@
+	flag.Parse()
+	suite.Run(t, new(NexusRequestForwardingSuite))
+}
+
+func (s *NexusRequestForwardingSuite) SetupSuite() {
+	s.dynamicConfigOverrides = map[dynamicconfig.Key]interface{}{
+		// Make sure we don't hit the rate limiter in tests
+		dynamicconfig.FrontendGlobalNamespaceNamespaceReplicationInducingAPIsRPS: 1000,
+		dynamicconfig.FrontendEnableNexusAPIs:                                    true,
+		nexusoperations.Enabled:                                                  true,
+		dynamicconfig.FrontendRefreshNexusIncomingServicesMinWait:                5 * time.Millisecond,
+	}
+	s.setupSuite([]string{"nexus_request_forwarding_active", "nexus_request_forwarding_standby"})
+}
+
+func (s *NexusRequestForwardingSuite) SetupTest() {
+	s.setupTest()
+}
+
+func (s *NexusRequestForwardingSuite) TearDownSuite() {
+	s.tearDownSuite()
+}
+
+func (s *NexusRequestForwardingSuite) TestStartOperationForwardedFromStandbyToActive() {
+	ctx := tests.NewContext()
+	namespace := fmt.Sprintf("%v-%v", "test-namespace", uuid.New())
+	taskQueue := fmt.Sprintf("%v-%v", "test-task-queue", uuid.New())
+	serviceName := fmt.Sprintf("%v-%v", "test-service", uuid.New())
+
+	activeFrontendClient := s.cluster1.GetFrontendClient()
+	regReq := &workflowservice.RegisterNamespaceRequest{
+		Namespace:                        namespace,
+		IsGlobalNamespace:                true,
+		Clusters:                         s.clusterReplicationConfig(),
+		ActiveClusterName:                s.clusterNames[0],
+		WorkflowExecutionRetentionPeriod: durationpb.New(7 * time.Hour * 24),
+	}
+	_, err := activeFrontendClient.RegisterNamespace(ctx, regReq)
+	s.NoError(err)
+
+	passiveFrontendClient := s.cluster2.GetFrontendClient()
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		// Wait for namespace record to be replicated.
+		_, err := passiveFrontendClient.DescribeNamespace(ctx, &workflowservice.DescribeNamespaceRequest{Namespace: namespace})
+		assert.NoError(t, err)
+	}, 15*time.Second, 500*time.Millisecond)
+
+	activeOperatorClient := s.cluster1.GetOperatorClient()
+	_, err = activeOperatorClient.CreateNexusIncomingService(ctx, &operatorservice.CreateNexusIncomingServiceRequest{
+		Spec: &nexuspb.IncomingServiceSpec{
+			Name:      serviceName,
+			Namespace: namespace,
+			TaskQueue: taskQueue,
+		},
+	})
+	s.NoError(err)
+
+	passiveOperatorClient := s.cluster2.GetOperatorClient()
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		_, err = passiveOperatorClient.CreateNexusIncomingService(ctx, &operatorservice.CreateNexusIncomingServiceRequest{
+			Spec: &nexuspb.IncomingServiceSpec{
+				Name:      serviceName,
+				Namespace: namespace,
+				TaskQueue: taskQueue,
+			},
+		})
+		assert.NoError(t, err)
+	}, 15*time.Second, 500*time.Millisecond)
+
+	dispatchURL := fmt.Sprintf("http://%s/%s", s.cluster2.GetHost().FrontendHTTPAddress(), cnexus.RouteDispatchNexusTaskByNamespaceAndTaskQueue.Path(cnexus.NamespaceAndTaskQueue{Namespace: namespace, TaskQueue: taskQueue}))
```

Comment:
If you're dispatching by namespace and task queue, why do you need to create an incoming service?